### DISCUSSION TITLE
Accessing protocol, host and port from plugin

### DISCUSSION THREAD

**Discussion by lewebsimple** [nuxt/nuxt#16972] [Questions]
Hello, I've searched quite a bit to figure this one out but I can't seem to find a clean way to access the resolved protocol, host and port values from within a Nuxt3 plugin.

The closest I got was using:
```
const host = process.env.NUXT_HOST || process.env.HOST || "localhost";
const port = process.env.NUXT_PORT || process.env.PORT || 3000;
console.log(`http://${host}:${port}/api/graphql`);
```

Anybody got an easier / more robust way of determining the full `/api/graphql` url above ?

**Comment by wobsoriano** [comment]
the `req` object now contains `x-forwarded-proto`, `x-forwarded-for`, and `host` that might be useful.

```js
export default defineNuxtPlugin((nuxtApp) => {
  const headers = useRequestHeaders()
  // headers['x-forwarded-for'] -> 127.0.0.1
  // headers['x-forwarded-proto'] -> IPv4
  // headers.host -> localhost:3000
})
```
